### DISCUSSION TITLE
Property 'user' does not exist on type '({ session: Session; user: AdapterUser; }

### DISCUSSION THREAD

**Discussion by RyanJMS** [nextauthjs/next-auth#9899] [Help] [answered]
I keep trying to get the user.id in callbacks, but it is saying property 'user' does not exist on type ({session: Session; user: AdapterUser;} ... maybe I'm missing something here, but even when I disable tslint on that line it will display the user id, can someone help me out?

This is my code.

import { DrizzleAdapter } from '@auth/drizzle-adapter'
import NextAuth, { DefaultSession } from 'next-auth'
import GitHub from 'next-auth/providers/github'
import google from 'next-auth/providers/google'
import { db } from '@/db'

declare module 'next-auth' {
    interface Session {
        user: {
            id: string
        } & DefaultSession['user']
    }
}

export const {
    handlers: { GET, POST },
    auth,
} = NextAuth({
    providers: [
        GitHub({ allowDangerousEmailAccountLinking: true }),
        google({ allowDangerousEmailAccountLinking: true }),
    ],
    session: {
        strategy: 'database',
    },
    adapter: DrizzleAdapter(db),
    callbacks: {
        async session({ session, user }) {
            if (session.user) {
                session.user.id = user.id
            }
            return session
        },
    },
})

**Comment by RyanJMS** [comment, marked as answer]
Seems to be a problem with 5.0.0-beta.5, downgraded to 5.0.0-beta.4 and it works as intended. Not sure if there's currently a fix for beta.5, but will keep looking.

**Comment by david-trujillo** [comment]
As workaround you can define the type like:

        async session({session, token}: { session: Session; token?: JWT, user?: User }) {

**Comment by donlair** [comment]
I'm having a similar problem and opened an issue at #9908

**Reply by RyanJMS** [reply]
Yeah I wasn't sure if I should open an issue, because I know they didn't want it being exposed due to security issues, and couldn't tell if it was a feature or a bug.
